### ISSUE TITLE
Add isnan(v) for fused_clamp kernel generated by jit codegen.

### ISSUE THREAD

**Issue by HangJie720** [pytorch/pytorch#64334]
For this program:

`import torch

torch._C._jit_set_profiling_executor(True)
torch._C._jit_set_profiling_mode(True)
torch._C._jit_override_can_fuse_on_cpu(True)
torch._C._jit_override_can_fuse_on_gpu(True)
torch._C._debug_set_fusion_group_inlining(False)
torch._C._jit_set_texpr_fuser_enabled(True)

def func2(a, b):
    return torch.clamp(a + b, min=0, max=2)
    #return torch.relu(a+b)

device = 'cuda'
a = torch.randn(4, 4, dtype=torch.float, device=device, requires_grad=True)
nan = torch.tensor(float('nan'), dtype=torch.float, device=device)

scripted_fn = torch.jit.script(func2)
script_outputs = scripted_fn(a,nan)
opt_script_outputs = scripted_fn(a,nan)

print(script_outputs.detach().cpu())
print(opt_script_outputs.detach().cpu())
`

this program will call 2 cuda kernel:
clamp_kernel_cuda is show as follow,
`void clamp_kernel_cuda(TensorIterator& iter, Scalar min_value, Scalar max_value) {
  AT_DISPATCH_ALL_TYPES_AND2(kHalf, kBFloat16, iter.dtype(), "clamp_cuda", [&]() {
    auto lower = min_value.to<scalar_t>();
    auto upper = max_value.to<scalar_t>();
    gpu_kernel(iter, [=]GPU_LAMBDA(scalar_t v) -> scalar_t {
      // Propagate nan, which doesn't propagate automatically for ROCm
      if (_isnan(v)) {
        return v;
      } else {
        return ::min(::max(v, lower), upper);
      }
    });
  });
}`

and jit codegen kernel is show as follow,

`extern "C" __global__
void fused_clamp(float* t0, float* aten_clamp) 
{
  if (512 * blockIdx.x + threadIdx.x<16 ? 1 : 0) {
    float t0_1 = t0[512 * blockIdx.x + threadIdx.x];
    aten_clamp[512 * blockIdx.x + threadIdx.x] = (t0_1<0.f ? 0.f : t0_1)>2.f ? 2.f : (t0_1<0.f ? 0.f : t0_1);
    }
  }
}`

My question is how to add nan value judgment for variable t0_1, expect generated kernel is show as follows,

`extern "C" __global__
void fused_clamp(float* t0, float* aten_clamp) 
{
  if (512 * blockIdx.x + threadIdx.x<16 ? 1 : 0) {
    float t0_1 = t0[512 * blockIdx.x + threadIdx.x];
    aten_clamp[512 * blockIdx.x + threadIdx.x] = isnan(t0_1) ? t0_1 : ((t0_1<0.f ? 0.f : t0_1)>2.f ? 2.f : (t0_1<0.f ? 0.f : t0_1));
    }
  }
}`

**Comment by janeyx99** [comment]
TorchScript has been deprecated in favor of torch.export as of 2.11. https://pytorch.org/blog/pytorch-2-11-release-blog/ Closing this issue as we are not further developing on it--please ping if you believe this issue was erroneously closed.
